pkg/nodeconfig/nodeconfig.go

```diff
@@ -255,25 +249,40 @@ func (nc *nodeConfig) configureNetwork() error {
 	return nil
 }
 
-// addVersionAnnotation adds the version annotation to nc.node
-func (nc *nodeConfig) addVersionAnnotation() {
-	nc.node.Annotations[VersionAnnotation] = version.Get()
-}
-
-// addPubKeyHashAnnotation adds the public key annotation to nc.node
-func (nc *nodeConfig) addPubKeyHashAnnotation() {
-	nc.node.Annotations[PubKeyHashAnnotation] = nc.publicKeyHash
+// addIdentifyingAnnotations adds the public key hash annotation as well as any additional annotations
+// that could help identify what type of node is being configured
+func (nc *nodeConfig) addIdentifyingAnnotations() error {
+	annotationsToAdd := map[string]string{PubKeyHashAnnotation: nc.publicKeyHash}
+	for key, value := range nc.additionalAnnotations {
+		annotationsToAdd[key] = value
+	}
+	return nc.addAnnotations(annotationsToAdd)
 }
 
-// addAdditionalAnnotations merges nc.additionalAnnotations into the annotations on nc.node. If the annotation
-// already existed, its value will be overwritten.
-func (nc *nodeConfig) addAdditionalAnnotations() {
-	if nc.additionalAnnotations == nil {
-		return
+// addAnnotations adds all the given annotations to node object and updates the nodeconfig
```

Comment:
Please name this `applyAnnotation` to be explicit as to what the method does and it syncs with k8s terminology. 

`updates the Node object in NodeConfig`